nodejs/awsx/ec2/vpc.ts

```diff
@@ -205,10 +233,122 @@ export class Vpc extends pulumi.ComponentResource {
      * This can be done by calling [subnet.createRoute] and passing in the newly created NatGateway.
      */
     public addNatGateway(name: string, args: x.ec2.NatGatewayArgs, opts: pulumi.ComponentResourceOptions = {}) {
-        const natGateway = new x.ec2.NatGateway(name, this, args, opts);
+        const natGateway = new x.ec2.NatGateway(name, this.parent, args, opts);
         this.natGateways.push(natGateway);
         return natGateway;
     }
+}
+
+(<any>VpcData).doNotCapture = true;
+utils.Capture(VpcData.prototype).addInternetGateway.doNotCapture = true;
+utils.Capture(VpcData.prototype).addNatGateway.doNotCapture = true;
+utils.Capture(VpcData.prototype).partition.doNotCapture = true;
+
+export class Vpc extends pulumi.ComponentResource {
+    public readonly id: pulumi.Output<string>;
+
+    public readonly vpc: pulumi.Output<aws.ec2.Vpc>;
+
+    private readonly _underlyingData: Promise<VpcData>;
+
+    constructor(name: string, args: VpcArgs | ExistingVpcArgs | ExistingVpcIdArgs, opts?: pulumi.ComponentResourceOptions);
+    /** @internal */
+    constructor(name: string, args: DefaultVpcArgs, opts?: pulumi.ComponentResourceOptions);
+    constructor(name: string, args: VpcArgs | ExistingVpcArgs | ExistingVpcIdArgs | DefaultVpcArgs, opts: pulumi.ComponentResourceOptions = {}) {
+        super("awsx:x:ec2:Vpc", name, {}, opts);
+
+        if (isExistingVpcArgs(args)) {
+            this._underlyingData = this.initializeExistingVpcArgs(name, args, opts);
+        }
+        else if (isDefaultVpcArgs(args)) {
+            this._underlyingData = this.initializeDefaultVpcArgs(name, args, opts);
+        }
+        else if (isExistingVpcIdArgs(args)) {
+            this._underlyingData = this.initializeExistingVpcIdArgs(name, args, opts);
+        }
+        else {
+            this._underlyingData = this.initializeVpcArgs(name, args, opts);
+        }
+
+        this.id = pulumi.output(this._underlyingData.then(v => v.id));
+        this.vpc = pulumi.output(this._underlyingData.then(d => d.vpc));
+
+        this._underlyingData.then(_ => this.registerOutputs());
+    }
+
+    protected isAsyncConstructed() {
+        return true;
+    }
+
+    /** @internal */
+    public async initializeExistingVpcArgs(name: string, args: ExistingVpcArgs, opts: pulumi.ComponentResourceOptions): Promise<VpcData> {
+        return new VpcData(name, this, args, opts);
+    }
+
+    /** @internal */
+    public async initializeExistingVpcIdArgs(name: string, args: ExistingVpcIdArgs, opts: pulumi.ComponentResourceOptions): Promise<VpcData> {
+        return new VpcData(name, this, args, opts);
+    }
+
+    /** @internal */
+    public async initializeDefaultVpcArgs(name: string, args: DefaultVpcArgs, opts: pulumi.ComponentResourceOptions): Promise<VpcData> {
```

Comment:
this is why we need `DefaultVpcArgs`.  So we could push the asynchrony of getting the default down to this level.